homeassistant/components/cert_expiry/sensor.py

```diff
@@ -56,106 +55,107 @@ def do_import(_):
+        return {
+            "is_valid": self.coordinator.is_cert_valid,
+            "error": str(self.coordinator.cert_error),
+        }
+
+
+class SSLCertificateDays(CertExpiryEntity):
+    """Implementation of the Cert Expiry days sensor."""
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return self._name
+        return f"Cert Expiry ({self.coordinator.name})"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if not self.coordinator.is_cert_valid:
+            return 0
+
+        if self.coordinator.data:
+            expiry = self.coordinator.data - dt.now()
+            return expiry.days
 
     @property
     def unique_id(self):
         """Return a unique id for the sensor."""
-        return f"{self.server_name}:{self.server_port}"
+        return f"{self.coordinator.host}:{self.coordinator.port}"
 
     @property
     def unit_of_measurement(self):
         """Return the unit this state is expressed in."""
         return TIME_DAYS
 
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        return self._state
 
-    @property
-    def icon(self):
-        """Icon to use in the frontend, if any."""
-        return "mdi:certificate"
+class SSLCertificateTimestamp(CertExpiryEntity):
+    """Implementation of the Cert Expiry timestamp sensor."""
 
     @property
-    def available(self):
-        """Return the availability of the sensor."""
-        return self._available
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return DEVICE_CLASS_TIMESTAMP
 
-    async def async_update(self):
-        """Fetch the certificate information."""
-        try:
-            days_to_expiry = await get_cert_time_to_expiry(
-                self.hass, self.server_name, self.server_port
-            )
-        except TemporaryFailure as err:
-            _LOGGER.error(err.args[0])
-            self._available = False
-            return
-        except ValidationFailure as err:
-            _LOGGER.error(
-                "Certificate validation error: %s [%s]", self.server_name, err
-            )
-            self._available = True
-            self._error = err
-            self._state = 0
-            self._valid = False
-            return
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception(
-                "Unknown error checking %s:%s", self.server_name, self.server_port
-            )
-            self._available = False
-            return
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"Cert Expiry Timestamp ({self.coordinator.name})"
 
-        self._available = True
-        self._error = None
-        self._state = days_to_expiry
-        self._valid = True
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self.coordinator.data
```

Comment:
The state should be isoformated UTC datetime.
```suggestion
        return self.coordinator.data.isoformat()
```